src/lib/rule2B_test.ts

```diff
@@ -0,0 +1,56 @@
+import {html, render} from 'lit-html';
+import {rule2B} from './rule2B';
+
+describe('The function for rule 2B', () => {
+  let container: HTMLElement;
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  it('returns null if the element has no aria-labelledby attribute', () => {
+    render(html`<div id="foo">Hello</div>`, container);
+    const elem = document.getElementById('foo');
+    expect(rule2B(elem!, {})).toBe(null);
+  });
+
+  it('returns null if the element has no valid aria-labelledby idrefs', () => {
+    render(html`<div id="foo" aria-labelledby="bar">Hello</div>`, container);
+    const elem = document.getElementById('foo');
+    expect(rule2B(elem!, {})).toBe(null);
+  });
+
+  it('returns concatenation of text alternatives of idreffed elements', () => {
+      render(
+        html`
+      <div id="foo" aria-labelledby="bar baz">Hello</div>
+      <div id="bar"></div>
+      <div id="baz"></div>
+    `,
+        container);
+    const elem = document.getElementById('foo');
+    expect(rule2B(elem!, {})).toBe('');
+  });
+
+  it('returns null if the node is already part of an aria-labelledby traversal',
+    () => {
+        render(
+          html`
+        <div id="foo" aria-labelledby="bar">Hello</div>
+        <div id="bar"></div>
+      `,
+          container);
+      const elem = document.getElementById('foo');
+      expect(rule2B(elem!, {ariaLabelledbyReference: true})).toBe(null);
+    });
+
+  /*
+   * Many more tests needed here but can't be verified yet due to lack
```

Comment:
add explicit `TODO` so that you remember to fill the test later on. 

It might be good to implement step 2G next so that you can write those tests.

Reply:
Good idea -- I'll implement 2F and 2G next (2F is required to reach 2G) as they will help test many other rules in the same way.